src/NotepadNext/dialogs/MainWindow.cpp

```diff
@@ -1,1989 +1,2005 @@
-    if (event->mimeData()->hasUrls() || event->mimeData()->hasText()) {
-        event->acceptProposedAction();
-    }
-    else {
-        event->ignore();
-    }
-}
-
-void MainWindow::dropEvent(QDropEvent *event)
-{
-    qInfo(Q_FUNC_INFO);
-
-    if (event->mimeData()->hasUrls()) {
-        // Get the urls into a stringlist
-        QStringList fileNames;
-        for (const QUrl &url : event->mimeData()->urls()) {
-            if (url.isLocalFile()) {
-                QFileInfo info(url.toLocalFile());
-
-                if (info.exists()) {
-                    if (info.isDir()) {
-                        QDirIterator it(url.toLocalFile(), QDir::Files, QDirIterator::FollowSymlinks| QDirIterator::Subdirectories);
-                        while (it.hasNext()) {
-                            fileNames << it.next();
-                        }
-                    }
-                    else {
-                        fileNames.append(url.toLocalFile());
-                    }
-                }
-            }
-        }
-
-        openFileList(fileNames);
-        bringWindowToForeground();
-        event->acceptProposedAction();
-    }
-    else if (event->mimeData()->hasText()) {
-        if (event->source()) {
-            // if it is from an editor, remove the text
-            ScintillaNext *sn = qobject_cast<ScintillaNext *>(event->source());
-            if (sn) {
-                sn->replaceSel("");
-            }
-        }
-
-        newFile();
-        currentEditor()->setText(event->mimeData()->text().toLocal8Bit().constData());
-        bringWindowToForeground();
-        event->acceptProposedAction();
-    }
-    else {
-        event->ignore();
-    }
-}
-
-void MainWindow::tabBarRightClicked(ScintillaNext *editor)
-{
-    qInfo(Q_FUNC_INFO);
-
-    // Focus on the correct tab
-    dockedEditor->switchToEditor(editor);
-
-    // Create the menu and show it
-    QMenu *menu = new QMenu(this);
-    menu->setAttribute(Qt::WA_DeleteOnClose);
-
-    menu->addAction(ui->actionClose);
-    menu->addAction(ui->actionCloseAllExceptActive);
-    menu->addAction(ui->actionCloseAllToLeft);
-    menu->addAction(ui->actionCloseAllToRight);
-    menu->addSeparator();
-    menu->addAction(ui->actionSave);
-    menu->addAction(ui->actionSaveAs);
-    menu->addAction(ui->actionRename);
-    menu->addSeparator();
-    menu->addAction(ui->actionReload);
-    menu->addSeparator();
-#ifdef Q_OS_WIN
-    menu->addAction(ui->actionShowInExplorer);
-    menu->addAction(ui->actionOpenCommandPromptHere);
-    menu->addSeparator();
-#endif
-    menu->addAction(ui->actionCopyFullPath);
-    menu->addAction(ui->actionCopyFileName);
-    menu->addAction(ui->actionCopyFileDirectory);
-    menu->popup(QCursor::pos());
-}
-
-void MainWindow::languageMenuTriggered()
-{
-    const QAction *act = qobject_cast<QAction *>(sender());
-    auto editor = currentEditor();
-    QVariant v = act->data();
-
-    setLanguage(editor, v.toString());
-}
+/*

```

Comment:
Well, your project mix LFCR file and LF togeter, some file use LF, and some use CR.
I'm also confused about this. How can I make git commit according to the different line breaks of each file in the repos?

Reply:
My suggestion is that the project owner performs an operation to unify the line breaks of the entire project.
For example, based on CRLF files that appear multiple times, perform a commit, and update the document to specify that all commits need to use CRLF. Otherwise, this problem will continue to occur in the future.
```
git ls-files -z | xargs -0 unix2dos
git config --global core.autocrlf true
git config --global core.safecrlf true
git commit -am "Change project repos all files with CRLF line breacks"
git push

```

Reply:
Weird. I've never ran into issues like this before. I'll have to look into it more to make sure the line endings are consistent.